src/api/java/CMakeLists.txt

```diff
@@ -9,20 +9,59 @@
 ## directory for licensing information.
 ##
 
-# create a directory for the cvc package
-file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/cvc")
+# copy cvc libraries to the resource directory
+file(GLOB CVC_LIBS "${CMAKE_INSTALL_PREFIX}/lib/*.so*")
+file(COPY ${CVC_LIBS} DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/src/main/resources")
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/cvc" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/src/main/java")
+file(COPY "/home/mudathir/Desktop/CVC4/mudathir/test/api/java/cvc" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/src/test/java")
+
+# copy gradle files
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/gradle" DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/gradlew" DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/build.gradle" DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
 # Generate cvc/Kind.java
 add_custom_target(
   gen-java-kinds
   ALL
   COMMAND
-    "${PYTHON_EXECUTABLE}"
-    "${CMAKE_CURRENT_LIST_DIR}/genkinds.py"
-    --kinds-header "${PROJECT_SOURCE_DIR}/src/api/cvc4cppkind.h"
-    --kinds-file-prefix "${CMAKE_CURRENT_BINARY_DIR}/cvc/Kind"
+  "${PYTHON_EXECUTABLE}"
+  "${CMAKE_CURRENT_LIST_DIR}/genkinds.py"
+  --kinds-header "${PROJECT_SOURCE_DIR}/src/api/cvc4cppkind.h"
+  --kinds-file-prefix "${CMAKE_CURRENT_BINARY_DIR}/src/main/java/cvc/Kind"
   DEPENDS
-    genkinds.py
+  genkinds.py
   COMMENT
-    "Generate Kind.java"
+  "Generate Kind.java"
 )
+
+
+find_package(Java)
+include(UseJava)
+find_package(JNI)
+
+message(STATUS "JNI_INCLUDE_DIRS=${JNI_INCLUDE_DIRS}")
+include_directories(${JNI_INCLUDE_DIRS})
```

Comment:
Do not use `include_directories` but `target_include_directories`.

Reply:
Ok

Reply:
Done

Reply:
I guess I misunderstood what is meant by target. It is now working.